library/ssl_tls.c

```diff
@@ -1921,6 +1921,52 @@ int mbedtls_ssl_set_hs_ecjpake_password( mbedtls_ssl_context *ssl,
 
     return( 0 );
 }
+
+int mbedtls_ssl_set_hs_ecjpake_password_opaque( mbedtls_ssl_context *ssl,
+                                                mbedtls_svc_key_id_t pwd )
+{
+    psa_pake_cipher_suite_t cipher_suite = psa_pake_cipher_suite_init();
+    psa_pake_role_t psa_role;
+    psa_status_t status;
+
+    if( ssl->handshake == NULL || ssl->conf == NULL )
+        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+
+    if( mbedtls_svc_key_id_is_null( pwd ) )
+        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+
+    psa_pake_cs_set_algorithm( &cipher_suite, PSA_ALG_JPAKE );
```

Comment:
I think the block of code between this line and `ssl->handshake->psa_pake_ctx_is_ok = 1;` included is common with `mbedtls_ssl_set_hs_ecjpake_password()` and could be turned into an auxiliary function. Then:

- `mbedtls_ssl_set_hs_ecjpake_password_opaque()` would validate inputs, call the auxiliary function, and `abort()` the operation on failure;
- `mbedtls_ssl_set_hs_ecjpake_password()` would validate inputs, create the key and store its handle in `ssl->handshake->psa_pake_password` (so that it can be `destroy()`ed when we free the handshake), call the auxiliary function, and on failure `abort()` the operation and `destroy()` the key.

Wdyt?